### PR TITLE
Mix client-side language to user while register (BKNDLSS-19178)

### PR DESCRIPTION
**Pull request by stanislav-grin**

### CODE REVIEW

File: src/users/register.js [hunk 1]
```diff
@@ -14,17 +14,47 @@ export function register(user /** async */) {
...
 
   return isAsync ? result : parseResponse(result)
 }
 
+function enrichWithLocaleInfo(user) {
+  if (user.blUserLocale === null || user.blUserLocale === '') {
+    return user
+  }
+
+  user.blUserLocale = getClientLanguage()
```
Comment (Valodya) [inline src/users/register.js]: also need to describe the property in the types definitions file https://github.com/Backendless/JS-SDK/blob/master/backendless.d.ts#L142

Comment (stanislav-grin) [reply src/users/register.js]: https://github.com/Backendless/JS-SDK/pull/138/commits/8236d55a663159a9ec52e0e73d762ab864c351ea

Comment (Valodya) [inline src/users/register.js]: what if user want to setup the `blUserLocale` property manually? look like it's not possible, isn't it?


what if we will detect/setup `blUserLocale` property only when there is no value
 
```
function enrichWithLocaleInfo(user) {
  if (!user.blUserLocale) {
   user.blUserLocale = getClientLanguage()
  }

return user
```

Comment (stanislav-grin) [reply src/users/register.js]: Changed next way:
https://github.com/Backendless/JS-SDK/pull/138/commits/8466f400b846aa39113143b59a161911ba7aeb6f

`user.blUserLocale` may be undefined, according to ticket description.

Comment (stanislav-grin) [reply src/users/register.js]: Fixed

File: src/users/register.js [hunk 2]
```diff
@@ -14,17 +14,45 @@ export function register(user /** async */) {
...
+  if (!user.blUserLocale) {
+    user.blUserLocale = getClientLanguage()
+  }
+
+  return user
+}
+
+function getClientLanguage() {
+  if (typeof navigator === 'undefined') {
+    return 'en'
```
[the remark this answers is not in the document]
Comment (stanislav-grin) [reply src/users/register.js]: Great idea

[the remark this answers is not in the document]
Comment (stanislav-grin) [reply src/users/register.js]: https://github.com/Backendless/JS-SDK/pull/138/commits/bc6118b5828a4a3da3a34931c21d547eeb98a793

**Pull request merged**